// ionic-angular/components/item/item-sliding.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import0 from 'ionic-angular/components/item/item-sliding';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view_utils';
import * as import3 from '@angular/core/src/linker/view';
import * as import5 from '@angular/core/src/linker/element';
import * as import6 from '@angular/core/src/linker/query_list';
import * as import8 from '@angular/core/src/linker/view_type';
import * as import9 from 'ionic-angular/components/list/list';
import * as import10 from '@angular/core/src/linker/element_ref';
import * as import11 from '@angular/core/src/zone/ng_zone';
import * as import12 from '@angular/core/src/metadata/view';
import * as import13 from '@angular/core/src/linker/component_factory';
export var Wrapper_ItemOptions = (function () {
    function Wrapper_ItemOptions(p0, p1) {
        this.changed = false;
        this._side = import1.UNINITIALIZED;
        this.context = new import0.ItemOptions(p0, p1);
    }
    Wrapper_ItemOptions.prototype.check_side = function (currValue, throwOnChange, forceUpdate) {
        if ((forceUpdate || import2.checkBinding(throwOnChange, this._side, currValue))) {
            this.changed = true;
            this.context.side = currValue;
            this._side = currValue;
        }
    };
    Wrapper_ItemOptions.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_ItemOptions;
}());
export var Wrapper_ItemSliding = (function () {
    function Wrapper_ItemSliding(p0, p1, p2, p3) {
        this.changed = false;
        this.context = new import0.ItemSliding(p0, p1, p2, p3);
    }
    Wrapper_ItemSliding.prototype.detectChangesInternal = function (view, el, throwOnChange) {
        var changed = this.changed;
        this.changed = false;
        return changed;
    };
    return Wrapper_ItemSliding;
}());
var renderType_ItemSliding_Host = null;
var _View_ItemSliding_Host0 = (function (_super) {
    __extends(_View_ItemSliding_Host0, _super);
    function _View_ItemSliding_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_ItemSliding_Host0, renderType_ItemSliding_Host, import8.ViewType.HOST, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckAlways);
    }
    _View_ItemSliding_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-item-sliding', rootSelector, null);
        this._appEl_0 = new import5.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_ItemSliding0(this.viewUtils, this.injector(0), this._appEl_0);
        this._ItemSliding_0_4 = new Wrapper_ItemSliding(this.parentInjector.get(import9.List, null), this.renderer, new import10.ElementRef(this._el_0), this.parentInjector.get(import11.NgZone));
        this._query_Item_0_0 = new import6.QueryList();
        this._query_ItemOptions_0_1 = new import6.QueryList();
        this._appEl_0.initComponent(this._ItemSliding_0_4.context, [], compView_0);
        this._query_Item_0_0.reset([]);
        this._ItemSliding_0_4.context.item = this._query_Item_0_0.first;
        compView_0.create(this._ItemSliding_0_4.context, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_ItemSliding_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import0.ItemSliding) && (0 === requestNodeIndex))) {
            return this._ItemSliding_0_4.context;
        }
        return notFoundResult;
    };
    _View_ItemSliding_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (this._ItemSliding_0_4.detectChangesInternal(this, this._el_0, throwOnChange)) {
            this._appEl_0.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if (this._query_ItemOptions_0_1.dirty) {
                this._query_ItemOptions_0_1.reset([]);
                this._ItemSliding_0_4.context._itemOptions = this._query_ItemOptions_0_1;
                this._query_ItemOptions_0_1.notifyOnChanges();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_ItemSliding_Host0;
}(import3.AppView));
function viewFactory_ItemSliding_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ItemSliding_Host === null)) {
        (renderType_ItemSliding_Host = viewUtils.createRenderComponentType('', 0, import12.ViewEncapsulation.None, [], {}));
    }
    return new _View_ItemSliding_Host0(viewUtils, parentInjector, declarationEl);
}
export var ItemSlidingNgFactory = new import13.ComponentFactory('ion-item-sliding', viewFactory_ItemSliding_Host0, import0.ItemSliding);
var styles_ItemSliding = [];
var renderType_ItemSliding = null;
var _View_ItemSliding0 = (function (_super) {
    __extends(_View_ItemSliding0, _super);
    function _View_ItemSliding0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_ItemSliding0, renderType_ItemSliding, import8.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import1.ChangeDetectorStatus.CheckOnce);
    }
    _View_ItemSliding0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._text_0 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this.renderer.projectNodes(parentRenderNode, import2.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._text_1 = this.renderer.createText(parentRenderNode, '\n    ', null);
        this.renderer.projectNodes(parentRenderNode, import2.flattenNestedViewRenderNodes(this.projectableNodes[1]));
        this._text_2 = this.renderer.createText(parentRenderNode, '\n  ', null);
        this.init([], [
            this._text_0,
            this._text_1,
            this._text_2
        ], [], []);
        return null;
    };
    return _View_ItemSliding0;
}(import3.AppView));
export function viewFactory_ItemSliding0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_ItemSliding === null)) {
        (renderType_ItemSliding = viewUtils.createRenderComponentType('', 2, import12.ViewEncapsulation.None, styles_ItemSliding, {}));
    }
    return new _View_ItemSliding0(viewUtils, parentInjector, declarationEl);
}
